log/DESIGN.md

```diff
@@ -0,0 +1,440 @@
+	SeverityFatal2
+	SeverityFatal3
+	SeverityFatal4
+)
+```
+
+[`SeverityText`](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitytext)
+is defined as `string` type.
+
+[`Body`](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-body)
+is defined as `string` type as the specification says:
+
+> First-party Applications SHOULD use a string message.
+
+[Log record attributes](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-attributes)
+are defined a regular slice of `attribute.KeyValue`.
+The users can use [`sync.Pool`](https://pkg.go.dev/sync#Pool)
+for reducing the number of allocations when passing attributes.
+
+Implementation requirements:
+
+- The [specification requires](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#concurrency-requirements)
+  the method to be safe to be called concurrently.
+
+- The method should not interrupt the record processing if the context is canceled
+  per ["ignoring context cancellation" guideline](../CONTRIBUTING.md#ignoring-context-cancellation).
+
+- The method handle the trace context passed via `ctx` argument in order to meet the
+  [specification's SDK requirement](https://opentelemetry.io/docs/specs/otel/logs/sdk/#readablelogrecord)
+  to populate the trace context fields from the resolved context.
+
+- The method should not modify the record's attribute as it may be reused by caller.
+
+- The method should copy the record's attributes in case of asynchronous processing
+  as otherwise it may lead to data races as the user may modify the passed attributes
+  e.g. because of using `sync.Pool` to reduce the number of allocation.
+
+- The [specification requires](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#emit-a-logrecord)
+  use the current time as observed timestamp if the passed is empty.
+
+`Emit` can be extended by adding new exported fields to the `Record` struct.
+
+Rejected alternatives:
+
+- [Record as interface](#record-as-interface)
+- [Options as parameter to Logger.Emit](#options-as-parameter-to-loggeremit)
+- [Passing record as pointer to Logger.Emit](#passing-record-as-pointer-to-loggeremit)
+- [Logger.WithAttributes](#loggerwithattributes)
+- [Record attributes like in slog.Record](#record-attributes-like-in-slogrecord)
+- [Record.Body as any](#recordbody-as-any)
+
+### noop package
+
+The `go.opentelemetry.io/otel/log/noop` package provides
+[Logs Bridge API No-Op Implementation](https://opentelemetry.io/docs/specs/otel/logs/noop/).
+
+### Trace context correlation
+
+The bridge implementation should do its best to pass
+the `ctx` containing the trace context from the caller
+so it can later passed via `Logger.Emit`.
+Re-constructing a `context.Context` with [`trace.ContextWithSpanContext`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#ContextWithSpanContext)
+and [`trace.NewSpanContext`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#NewSpanContext)
+would usually involve more memory allocations.
```

Comment:
I'm not sure I follow what this is trying to say. Is it saying that we won't construct a context for users?

Reply:
Yes. The `context.Context` should be passed by the user.

Reply:
Maybe something like "It is not expected that users reconstruct a `context.Context` ..."

Reply:
PTAL 3f25db7c